x/examples/soax-relay/README.md

```diff
@@ -0,0 +1,131 @@
+# SOCKS5 Proxy Relay for SOAX
+
+This program creates a SOCKS5 proxy relay that allows users to share access to SOAX without sharing their own credentials. It's designed for connectivity testing purposes.
+
+## Design
+
+The proxy relay works as follows:
+
+1. It listens for incoming SOCKS5 connections on a specified local address and port.
+2. It authenticates incoming connections using a custom authentication mechanism.
+3. For authenticated connections, it relays requests to the SOAX proxy server.
+4. It supports both TCP and UDP traffic.
+
+#### Key components:
+
+- **CustomAuthenticator**: Handles authentication based on username prefix and extracts the suffix which includes connection parameters such as country, city, ISP, etc.
+- **StaticCredentialStore**: Stores and validates user credentials.
+- **UDP Associate Handler**: Manages UDP connections by relaying the SOAX bind address back to the client.
+
+## Usage
+
+### Configuration
+
+The program uses a YAML configuration file (`config.yaml`) for settings. Example configuration:
+
+```yaml
+server:
+  address: "127.0.0.1"
+  port: "1080"
+
+upstream:
+  prefix: "package-xxxxx-"
+  password: "YourSOAXPassword"
+  address: "proxy.soax.com:5000"
```

Comment:
Let's simplify this and be explicit about the soax info, using the [names they use](https://helpcenter.soax.com/en/articles/6723733-sticky-sessions):

```suggestion
soax:
  package_id: "XXXXXX"
  package_key: "YourSOAXPassword"
```

I'd omit the server address too, it's easier.